src/operator/contrib/ngraph-inl.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2018 Intel Corporation
+ * \file ngraph.cc
+ * \brief ngraph subgraph property for mxnet
+*/
+
+#ifndef MXNET_OPERATOR_CONTRIB_NGRAPH_INL_H_
+#define MXNET_OPERATOR_CONTRIB_NGRAPH_INL_H_
+
+#if MXNET_USE_NGRAPH
+#include <ngraph_compiler.h>
+#include <ngraph_imperative.h>
+#include <ngraph_nnvm_ops.h>
+#include <ngraph_utils.h>
+
+#include <vector>
+
+#include "../subgraph/common.h"
+#include "../subgraph/subgraph_property.h"
+
+namespace mxnet {
+namespace op {
+
+// when built with NGRAPH we use this subgraph by default
+static int ngraph_backend = setenv("MXNET_SUBGRAPH_BACKEND", "ngraph", 0);
```

Comment:
should we use nGraph by default when ngraph is built into mxnet?

Reply:
Since you have to build MXNet from source to enable this experimental feature, we thought it made sense to enable it by default for experimental runs. If you disagree, it's easy enough to turn off.

Reply:
This also forces all CI/tests by default to exercise `MXNET_SUBGRAPH_BACKEND=ngraph` , when mxnet is build with `USE_NGRAPH`. User is expected to use this subgraph (`ngraph`), if they build with it.

Reply:
to be honest, I still feel this is not a good idea to make ngraph a default backend when ngraph is built in mxnet.

Reply:
we will remove it